hawkbit-core/src/test/java/org/eclipse/hawkbit/api/PropertyBasedArtifactUrlHandlerTest.java

```diff
@@ -0,0 +1,118 @@
+    private static final String sha1Hash = "test12345";
+    private static final Long artifactId = 1345678L;
+    private static final String TENANT = "TEST_TENANT";
+
+    private static URLPlaceholder placeholder = new URLPlaceholder(TENANT, TENANT_ID, CONTROLLER_ID, targetId,
+            new SoftwareData(softwareModuleId, fileName, artifactId, sha1Hash));
+
+    @Before
+    public void setup() {
+        properties = new ArtifactUrlHandlerProperties();
+        urlHandlerUnderTest = new PropertyBasedArtifactUrlHandler(properties);
+
+    }
+
+    @Test
+    @Description("Tests the generation of http download url.")
+    public void urlGenerationWithDefaultConfiguration() {
+        properties.getProtocols().put("download-http", new UrlProtocol());
+
+        final List<ArtifactUrl> ddiUrls = urlHandlerUnderTest.getUrls(placeholder, APIType.DDI);
+        assertEquals(
+                Lists.newArrayList(new ArtifactUrl("http", "download-http", HTTP_LOCALHOST + TENANT + "/controller/v1/"
+                        + CONTROLLER_ID + "/softwaremodules/" + softwareModuleId + "/artifacts/" + fileName)),
+                ddiUrls);
+
+        final List<ArtifactUrl> dmfUrls = urlHandlerUnderTest.getUrls(placeholder, APIType.DMF);
+        assertEquals(ddiUrls, dmfUrls);
+    }
+
+    @Test
+    @Description("Tests the generation of custom download url with a CoAP example that supports DMF only.")
+    public void urlGenerationWithCustomConfiguration() {
+        final UrlProtocol proto = new UrlProtocol();
+        proto.setIp("127.0.0.1");
+        proto.setPort(5683);
+        proto.setProtocol("coap");
+        proto.setRel("coap");
+        proto.setSupports(Lists.newArrayList(APIType.DMF));
+        proto.setRef("{protocol}://{ip}:{port}/fw/{tenant}/{controllerId}/sha1/{artifactSHA1}");
+        properties.getProtocols().put("coap", proto);
+
+        List<ArtifactUrl> urls = urlHandlerUnderTest.getUrls(placeholder, APIType.DDI);
+
+        assertThat(urls).isEmpty();
+        urls = urlHandlerUnderTest.getUrls(placeholder, APIType.DMF);
+
+        assertEquals(Lists.newArrayList(new ArtifactUrl("coap", "coap",
+                "coap://127.0.0.1:5683/fw/" + TENANT + "/" + CONTROLLER_ID + "/sha1/" + sha1Hash)), urls);
+    }
+
+    @Test
+    @Description("Tests the generation of custom download url with a CoAP example that supports DMF only.")
+    public void urlGenerationWithCustomShortConfiguration() {
+        final UrlProtocol proto = new UrlProtocol();
+        proto.setIp("127.0.0.1");
+        proto.setPort(5683);
+        proto.setProtocol("ftp");
+        proto.setRel("ftp");
+        proto.setSupports(Lists.newArrayList(APIType.DMF));
+        proto.setRef("{protocol}://{ip}:{port}/fws/{tenant}/{targetIdBase62}/{artifactIdBase62}");
+        properties.getProtocols().put("ftp", proto);
+
+        List<ArtifactUrl> urls = urlHandlerUnderTest.getUrls(placeholder, APIType.DDI);
+
+        assertThat(urls).isEmpty();
```

Comment:
What about the other way round? DDI is supporting protocol that is not handled by DMF? Is this possible or the same behavior?